### PR TITLE
Whitelisting PatWild for E0022 (fixes #28822)

### LINKED ISSUE #28822: const fn with `_` as an arg name vomits
## Doesn't work:

```rust
#![feature(const_fn)]
fn main(){}
const fn size_ofs(_: usize) { }
```

```
<anon>:3:19: 3:20 error: arguments of constant functions can only be immutable by-value bindings [E0022]
<anon>:3 const fn size_ofs(_: usize) { }
                           ^
<anon>:3:19: 3:20 help: see the detailed explanation for E0022
error: aborting due to previous error
playpen: application terminated with error code 101
```

-----

## Works:

```rust
#![feature(const_fn)]
fn main(){}
const fn size_ofs(_foo: usize) { }
```

```
<anon>:3:1: 3:35 warning: function is never used: `size_ofs`, #[warn(dead_code)] on by default
<anon>:3 const fn size_ofs(_foo: usize) { }
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

### PR DESCRIPTION
**Pull request by GlenDC**
Here is my attempt to resolve issue #28822, @Manishearth.
Please let me know if it's fine. And if not, what should I do instead?

This issue felt like quite a good start for some rust contributions. It allows me to get used to the workflow  and codebase of rust in an easy-to-swallow manner. Are there any other issues you would recommend me to look at? :) Would love to do some more stuff!

### CODE REVIEW

File: src/test/run-pass/issue-28822.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(const_fn)]
+
+fn main() {}
+
+const fn size_ofs(_: usize) {}
+const fn size_ofs2(_foo: usize) {}
```
Comment (Manishearth) [inline src/test/run-pass/issue-28822.rs]: nit: there should be a newline at the end of the file

**Comment by Manishearth** [comment]
Looks good to me! Minor nit and we can land this. (Just use ``git commit --amend`, no need to create a new commit)

I usually have a "second bug" in my pocket for people to work on, but I'm a bit busy with school stuff right now so it might take me a day or two to get to this. In the meantime, you can find easy issues [here](https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AE-easy), https://github.com/rust-lang/rust/issues/28518 looks pretty easy since it's just tests. I'll look for something substantial but easy in the compiler for you to work on in a few days (remind me if I forget)

**Comment by GlenDC** [comment]
Ok @Manishearth I (force) updated the current commit. It has the newline amended to it. Thanks for the help. I will take a look at #28518 later today :) Let me know if something is wrong with this pull request and in case you find more issues you think are suitable for me :)

**Comment by Manishearth** [comment]
@bors r+ rollup

LGTM, thanks!

**Pull request merged**